packages/react-core/src/hooks/useUserAttributes/__tests__/useUserAttributes.test.ts

```diff
@@ -0,0 +1,379 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+
+import {
+  SendUserAttributeVerificationCodeOutput,
+  UpdateUserAttributesOutput,
+  UserAttributeKey,
+  confirmUserAttribute,
+  deleteUserAttributes,
+  fetchUserAttributes,
+  sendUserAttributeVerificationCode,
+  updateUserAttributes,
+} from '@aws-amplify/auth';
+
+import * as AuthModule from '@aws-amplify/auth';
+import {
+  ConfirmInput,
+  DeleteInput,
+  FetchInput,
+  HandleAttributeActionOutput,
+  SendCodeInput,
+  UpdateInput,
+  VerifiableAttribute,
+} from '../types';
+import useUserAttributes from '../useUserAttributes';
+
+const confirmUserAttributeSpy = jest.spyOn(AuthModule, 'confirmUserAttribute');
+const deleteUserAttributesSpy = jest.spyOn(AuthModule, 'deleteUserAttributes');
+const fetchUserAttributesSpy = jest.spyOn(AuthModule, 'fetchUserAttributes');
+const sendCodeSpy = jest.spyOn(AuthModule, 'sendUserAttributeVerificationCode');
+const updateUserAttributesSpy = jest.spyOn(AuthModule, 'updateUserAttributes');
+
+const mockAttributes: { [Attribute in UserAttributeKey]?: string } = {
```

Comment:
This is not really a "mock" - generally a "mock" is a function that can be controlled to provide manually defined results to assert test behavior. Would name these based on their purpose, e.g. `mockAttributes` -> `fetchUserAttributesResult`, etc